modules/tree/src/main/tree.scala

```diff
@@ -1,16 +1,158 @@
 package lila.tree
 
+import alleycats.Zero
 import chess.Centis
 import chess.format.pgn.{ Glyph, Glyphs }
-import chess.format.{ Fen, Uci, UciCharPair }
+import chess.format.{ Fen, Uci, UciCharPair, UciPath }
 import chess.opening.Opening
 import chess.{ Ply, Square, Check }
-import chess.variant.Crazyhouse
+import chess.variant.{Variant, Crazyhouse}
 import play.api.libs.json.*
 import ornicar.scalalib.ThreadLocalRandom
 
 import lila.common.Json.{ *, given }
 
+import Node.{Comments, Comment, Gamebook, Shapes}
+
+//opaque type not working due to cyclic ref try again later
+// either we decide that branches strictly represent all the children from a node
+// with the first being the mainline, OR we just use it as an List with extra functionalities 
+case class Branches(nodes: List[Branch]) extends AnyVal:
```

Comment:
Couldn't make it work with an opaque type

Reply:
Also should we stick to the current terminology? `Node` as the trait, `Branch` as non-root Nodes, and `Branches`